content-repo/gendocs.py

```diff
@@ -457,6 +485,120 @@ def insert_approved_tags_and_usecases():
+    """Get the deprecation start date. The 1st of the following month.
+
+    Args:
+        dep_date (datetime): The raw dep date
+
+    Returns:
+        tuple of start deprecation and end deprecation
+    """
+    DATE_FRMT = "%b %d, %Y"
+    start = datetime(day=1, month=dep_date.month, year=dep_date.year) + relativedelta(months=+1)
+    end = start + relativedelta(months=+6)
+    return (datetime.strftime(start, DATE_FRMT), datetime.strftime(end, DATE_FRMT))
+
+
+def find_deprecated_integrations(content_dir: str):
+    files = glob.glob(content_dir + '/Packs/*/Integrations/*.yml')
+    files.extend(glob.glob(content_dir + '/Packs/*/Integrations/*/*.yml'))
+    res: List[DeprecatedInfo] = []
+    # go over each file and check if contains deprecated: true
+    for f in files:
+        with open(f, 'r') as fr:
+            content = fr.read()
+            if dep_search  := re.search(r'^deprecated:\s*true', content, re.MULTILINE):
+                pack_dir = re.match(r'.+/Packs/.+?(?=/)', f)
+                if is_xsoar_supported_pack(pack_dir.group(0)):  # type: ignore[union-attr]
+                    yml_data = yaml.safe_load(content)
+                    id = yml_data.get('commonfields', {}).get('id') or yml_data['name']
+                    name: str = yml_data.get('display') or yml_data['name']
+                    desc = yml_data.get('description')
+                    content_to_search = content[:dep_search.regs[0][0]]
+                    lines_search = re.findall(r'\n', content_to_search)
+                    blame_line = 1
+                    if lines_search:
+                        blame_line += len(lines_search)
+                    dep_date = get_blame_date(content_dir, f, blame_line)
+                    maintenance_start, eol_start = get_deprecated_display_dates(dep_date)
+                    dep_suffix = "(Deprecated)"
+                    if name.endswith(dep_suffix):
+                        name = name.replace(dep_suffix, "").strip()
+                    info = DeprecatedInfo(id=id, name=name, description=desc, note=get_extracted_deprecated_note(desc),
+                                          maintenance_start=maintenance_start, eol_start=eol_start)
+                    print(f'Adding deprecated integration: [{name}]. Deprecated date: {dep_date}. From file: {f}')
+                    res.append(info)
+                else:
+                    print(f'Skippinng deprecated integration: {f} which is not supported by xsoar')
+    return res
+
+
+def merge_deprecated_info(deprecated_list: List[DeprecatedInfo], deperecated_info_file: str):
+    with open(deperecated_info_file, "rt") as f:
+        to_merge_list: List[DeprecatedInfo] = json.load(f)['integrations']
+    to_merge_map = {i['id']: i for i in to_merge_list}
+    merged_list: List[DeprecatedInfo] = []
+    for d in deprecated_list:
+        if d['id'] in to_merge_map:
+            d = {**d, **to_merge_map[d['id']]}  # type: ignore
```

Comment:
please add error code here

Reply:
It is a known limitation of mypy when merging dicts. https://github.com/python/mypy/issues/4122. Will add the error code though for completeness.